src/Workspaces/Core/Portable/Execution/PinnedRemotableDataScope.cs

```diff
@@ -10,15 +10,46 @@
 namespace Microsoft.CodeAnalysis.Execution
 {
     /// <summary>
+    /// Information related to pinned solution
+    /// </summary>
+    internal class PinnedSolutionInfo
+    {
+        /// <summary>
+        /// Unique ID for this pinned solution
+        /// 
+        /// This later used to find matching solution between VS and remote host
+        /// </summary>
+        public readonly int ScopeId;
+
+        /// <summary>
+        /// This indicates whether this scope is for primary branch or not (not forked solution)
+        /// 
+        /// Features like OOP will use this flag to see whether caching information related to this solution
+        /// can benefit other requests or not
+        /// </summary>
```

Comment:
OOP has 2 lever of caching.

one is primary solution like our Workspace.CurrentSolution. so next solution can be created incrementally over current solution. we never hold onto those forked solution even in VS. we actually never saved any information on forked solution in any form. they lives in memory and go away.

the other is just pure cache. it holds onto (checksum, data) for 3 min. and let them go. so forked solution case, it cost shouldn't be any different than how things happened in VS